### PR TITLE
Add email workers support

### PR DESCRIPTION
**Pull request by nelsonjsduarte**
What this PR solves / how to test:
Create Email worker on init worker setup

1. wrangler init <project-name>
2. Select "Email handler"

Associated docs issues/PR:
- [x] https://developers.cloudflare.com/email-routing/email-workers/

Author has included the following, where applicable:

- [ ] Tests
- [X] Changeset

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

Internal Ticket

### CODE REVIEW

File: packages/wrangler/templates/new-worker-email.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export interface Env {
+	// Example binding to KV. Learn more at https://developers.cloudflare.com/workers/runtime-apis/kv/
+	// MY_KV_NAMESPACE: KVNamespace;
+	//
+	// Example binding to Durable Object. Learn more at https://developers.cloudflare.com/workers/runtime-apis/durable-objects/
+	// MY_DURABLE_OBJECT: DurableObjectNamespace;
+	//
+	// Example binding to R2. Learn more at https://developers.cloudflare.com/workers/runtime-apis/r2/
+	// MY_BUCKET: R2Bucket;
+}
```
Comment (JacobMGEvans) [inline packages/wrangler/templates/new-worker-email.ts:16]: This can be generated with `wrangler types` which creates a DTS file for the user.

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
Not sure the `interface` is necessary. I am not firm on this though.

File: packages/wrangler/templates/new-worker-email.ts (hunk 1 above)
Comment (rozenmd) [reply packages/wrangler/templates/new-worker-email.ts:16]: The point of this mostly commented-out `interface` is to let folks know about other bindings they can use, FWIW

Comment (JacobMGEvans) [reply packages/wrangler/templates/new-worker-email.ts:16]: Alright, I won't block on this.

**Review by JacobMGEvans — APPROVED** [review]

**Comment by lrapoport-cf** [comment]
hey @nelsonjsduarte :) did you still want to get this PR landed, or should it be closed?

**Comment by jspspike** [comment]
Closing this for now, feel free to open this later if you would like to continue with this PR. Thanks!

**Pull request closed without merging**